<!-- Start:AutoDoc:: Modify `src/readme/rules.ts` and run `gulp readme` to update block -->
## no-invalid-regexp (ESLint: [no-invalid-regexp](http://eslint.org/docs/rules/no-invalid-regexp))
[![rule_source](https://img.shields.io/badge/%F0%9F%93%8F%20rule-source-green.svg)](https://github.com/buzinas/tslint-eslint-rules/blob/master/src/rules/noInvalidRegexpRule.ts)
[![test_source](https://img.shields.io/badge/%F0%9F%93%98%20test-source-blue.svg)](https://github.com/buzinas/tslint-eslint-rules/blob/master/src/test/rules/noInvalidRegexpRuleTests.ts)

disallow invalid regular expression strings in the `RegExp` constructor (recommended)

### Usage

```json
"no-invalid-regexp": true
```

<!-- End:AutoDoc -->
