<!-- Start:AutoDoc:: Modify `src/readme/rules.ts` and run `gulp readme` to update block -->
## valid-typeof (ESLint: [valid-typeof](http://eslint.org/docs/rules/valid-typeof))
[![rule_source](https://img.shields.io/badge/%F0%9F%93%8F%20rule-source-green.svg)](https://github.com/buzinas/tslint-eslint-rules/blob/master/src/rules/validTypeofRule.ts)
[![test_source](https://img.shields.io/badge/%F0%9F%93%98%20test-source-blue.svg)](https://github.com/buzinas/tslint-eslint-rules/blob/master/src/test/rules/validTypeofRuleTests.ts)

Ensure that the results of typeof are compared against a valid string (recommended)

### Usage

```json
"valid-typeof": true
```

<!-- End:AutoDoc -->
